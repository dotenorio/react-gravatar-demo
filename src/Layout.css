* { box-sizing: border-box; margin: 0; padding: 0 }
html,
body {
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}

.header {
  background-color: #eee;
  padding: 25px;
  box-shadow: 0 1px 10px #999;
  display: flex;
}

.header h1 {
  margin-right: 25px;
}

.header ul {
  padding: 0px;
  margin: 0px;
  list-style: none;  
  align-self: center;
  padding-top: 5px;
}

.header ul li {
  display: inline;
}

.header ul li a {
  padding: 2px 10px;
  display: inline-block;
  color: #106274;
  text-decoration: none;
}

.header ul li a:hover {
  opacity: 0.6;
}

.header ul li a.active {
  font-weight: bold;
}

.content {
  margin: 40px 25px;
}

@media only screen and (max-width: 375px) {
  .content {
    margin: 40px 15px;
  }
  .header h1 {
    font-size: 150%;
  }
}

@media only screen and (max-width: 768px) {
  .header { 
    display: block;
  }
  .header h1 {
    text-align: center;
    margin-right: 0;
    margin-bottom: 15px;
  }
  .header ul {
    text-align: center;
  }
}