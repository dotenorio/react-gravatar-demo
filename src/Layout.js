import React from 'react';
import { NavLink } from "react-router-dom";

import './Layout.css';

function Header(props) {
  return (
    <div className="header">
      <h1>
        {props.title}
      </h1>
      <ul>
        <li>
          <NavLink to="/" exact activeClassName="active" title="Home Page">Home</NavLink>
        </li>
        <li>
          <NavLink to="/about" activeClassName="active" title="About Project">About</NavLink>
        </li>
      </ul>
    </div>
  );
}

function Layout (props) {
  return (
    <div>
      <Header title={ props.title ? props.title : "React Gravatar Demo"} />
      <div className="content">
        {props.children}
      </div>
    </div>
  );
}

export default Layout;