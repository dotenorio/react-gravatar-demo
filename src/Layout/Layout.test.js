import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import Layout from './Layout';

test('Load layout with title and children', () => {
  const title = 'Test Title'
  const content = 'Test'
  const wrapper = shallow(
    <Layout title={title}>
      <p>{content}</p>
    </Layout>
  );

  expect(wrapper.find('div.content').exists()).toEqual(true);
  expect(wrapper.find('Header').props().title).toEqual(title);
  expect(wrapper.contains(<p>{content}</p>)).toEqual(true);
})

test('Layout renders correctly', () => {
  const tree = renderer
    .create(<MemoryRouter><Layout /></MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});