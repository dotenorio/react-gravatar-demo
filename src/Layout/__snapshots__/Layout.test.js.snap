// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div>
  <div
    className="header"
  >
    <h1>
      React Gravatar Demo
    </h1>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={Object {}}
          title="Home Page"
        >
          Home
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/about"
          onClick={[Function]}
          title="About Project"
        >
          About
        </a>
      </li>
    </ul>
  </div>
  <div
    className="content"
  />
</div>
`;
